qcore/timeseries.py

```diff
@@ -502,29 +505,47 @@ def vel2txt(self, station, prefix="./", title="", dt=None, acc=False):
         else:
             f = self.vel
         for i, c in enumerate(f(station, dt=dt).T):
-            seis2txt(
+            timeseries_to_text(
                 c,
+                f"{prefix}{station}.{self.COMP_NAME[i]}",
                 dt,
-                prefix,
                 station,
                 self.COMP_NAME[i],
                 start_sec=self.start_sec,
                 title=title,
             )
 
-    def all2txt(self, prefix="./", dt=None, f="vel"):
+    def all2txt(self, prefix: str = "./", dt: float = None, f: str = "vel"):
         """
-        NOTE: This function is not designed to be used other than for single/debug use.
+        Creates waveforms in text files for all stations.
+        Note: This function is not designed to be used other than for single/debug use.
         Make a parallel wrapper for any "real" use cases.
         Produces text files previously done by script called `winbin-aio`.
         For compatibility. Consecutive file indexes in parallel for performance.
         Slowest part is numpy formating numbers into text and number of lines.
+
+        Parameters
+        ----------
+        prefix : str, optional, default="./"
+            The prefix is an output path combined with an optional filename prefix.
+            eg. prefix = "dir1/dir2/XXX", prefix_filename = "XXX" and prefix_dirname = "dir1/dir2"
+            eg. prefix = "dir1/dir2/", prefix_filename = "" and prefix_dirname = "dir1/dir2"
+        dt : float, optional, default=None
+            The time step of the data
+        f : str, optional, default="vel"
+            The type of data to save. Options are "acc" and "vel"
         """
+
         if dt is None:
             dt = self.dt
         acc = f == "acc"
+
+        prefix_chunks = prefix.split("/")
+        prefix_filename = prefix_chunks[-1]
+        prefix_dirname = Path("/".join(prefix_chunks[:-1])).resolve()
+        prefix_dirname.mkdir(parents=True, exist_ok=True)
         for s in self.stations.name:
```

Comment:
I'd simplify this to something like:

```suggestion
        prefix_path = Path(prefix)
        prefix_path.parent.mkdir(parents=True, exist_ok=True)
        for s in self.stations.name:
            self.vel2txt(s, prefix=prefix, title=prefix_path.stem, dt=dt, acc=acc)
```

Reply:
You may have to `resolve` there again idk.

Reply:
It is slightly trickier.
Consider prefix "AA/BB/CC" vs "AA/BB/CC/".
The former means in AA/BB directory, files with CC prefix eg. CC0000, CC0001...
The latter means in AA/BB/CC directory, files with empty prefix, so 0000, 0001 ...
Your suggested change will treat both cases as the former case.